src/HigherOrderComponent/DropTargetMap/DropTargetMap.example.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { render } from 'react-dom';
+import OlSourceOsm from 'ol/source/osm';
+import OlLayerTile from 'ol/layer/tile';
+import OlView from 'ol/view';
+import OlMap from 'ol/map';
+
+import {
+  MapComponent,
+  MapProvider,
+  mappify,
+  onDropAware
+} from '../../index.js';
+
+/**
+ * Create the OlMap (you could do some asynchronus stuff here).
+ *
+ * @return {Promise} Promise that resolves an OlMap.
+ */
+const mapPromise = new Promise((resolve) => {
+  const layer = new OlLayerTile({
+    source: new OlSourceOsm()
+  });
+
+  const map = new OlMap({
+    view: new OlView({
+      center: [
+        135.1691495,
+        34.6565482
+      ],
+      projection: 'EPSG:4326',
+      zoom: 16,
+    }),
+    layers: [layer]
+  });
+
+  resolve(map);
+});
+
+const Map = mappify(onDropAware(MapComponent));
```

Comment:
Is the order here relevant?

Reply:
I don't think so. The resulting map component will have the onDrop/onDragOver and map properties injected by all wrapped HOCs and the map is only used once the user actually drops a file.